server/endpoint/endpoint.go

```diff
@@ -0,0 +1,73 @@
+package endpoint
+
+import (
+	"github.com/giantswarm/microendpoint/endpoint/healthz"
+	"github.com/giantswarm/microendpoint/endpoint/version"
+	healthzservice "github.com/giantswarm/microendpoint/service/healthz"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/node-operator/server/middleware"
+	"github.com/giantswarm/node-operator/service"
+)
+
+// Config represents the configuration used to create a endpoint.
+type Config struct {
+	// Dependencies.
+	Logger     micrologger.Logger
+	Middleware *middleware.Middleware
+	Service    *service.Service
+}
+
+// DefaultConfig provides a default configuration to create a new endpoint by
+// best effort.
+func DefaultConfig() Config {
```

Comment:
I think we agreed to drop these `DefaultConfig()` functions because they don't produce a working config and most of the time are only empty values (e.g. here it could be `return Config{}`).

Reply:
I dropped the config constructor where I found it and where it was easy.